server/src/main/scala/ch.epfl.scala.index.server/DataRepository.scala

```diff
@@ -1,667 +1,615 @@
+          .map { prefix =>
+            dismax(
+              prefixQuery("repository", prefix).boost(6),
+              prefixQuery("primaryTopic", prefix).boost(5),
+              prefixQuery("organization", prefix).boost(5),
+              prefixQuery("github.description", prefix).boost(4),
+              prefixQuery("github.topics", prefix).boost(4),
+              prefixQuery("artifacts", prefix).boost(2)
+            )
           }
-        }),
-      Duration.Inf
+          .getOrElse(matchNoneQuery())
+
+        should(
+          multiMatch,
+          readmeMatch,
+          autocompleteQuery,
+          contributingQuery
+        )
+      }
+    }
+
+    must(filterQuery, plainTextQuery)
+  }
+
+  private def topicsQuery(topics: Seq[String]): QueryDefinition = {
+    must(topics.map(topicQuery))
+  }
+
+  private def topicQuery(topic: String): QueryDefinition = {
+    termQuery("github.topics", topic)
+  }
+
+  private val cliQuery = termQuery("hasCli", true)
+
+  private def repositoriesQuery(
+      repositories: Seq[GithubRepo]
+  ): QueryDefinition = {
+    should(repositories.map(repositoryQuery))
+  }
+
+  private def repositoryQuery(repo: GithubRepo): QueryDefinition = {
+    must(
+      termQuery("organization", repo.organization),
+      termQuery("repository", repo.repository)
+    )
+  }
+
+  private def targetsQuery(
+      targetTypes: Seq[String],
+      scalaVersions: Seq[String],
+      scalaJsVersions: Seq[String],
+      scalaNativeVersions: Seq[String],
+      sbtVersions: Seq[String]
+  ): QueryDefinition = {
+    must(
+      targetTypes.map(termQuery("targetType", _)) ++
+        scalaVersions.map(termQuery("scalaVersion", _)) ++
+        scalaJsVersions.map(termQuery("scalaJsVersion", _)) ++
+        scalaNativeVersions.map(termQuery("scalaNativeVersion", _)) ++
+        sbtVersions.map(termQuery("sbtVersion", _))
     )
   }
 
+  private def targetQuery(target: ScalaTarget): QueryDefinition = {
+    must(
+      termQuery("scalaVersion", target.scalaVersion.toString),
+      optionalQuery(target.scalaJsVersion) { version =>
+        termQuery("scalaJsVersion", version.toString)
+      },
+      optionalQuery(target.scalaNativeVersion) { version =>
+        termQuery("scalaNativeVersion", version.toString)
+      },
+      optionalQuery(target.sbtVersion) { version =>
+        termQuery("sbtVersion", version.toString)
+      }
+    )
+  }
+
+  private val contributingQuery = boolQuery().must(
+    Seq(
+      nestedQuery(
+        "github.beginnerIssues",
+        existsQuery("github.beginnerIssues")
+      ),
+      existsQuery("github.contributingGuide"),
+      existsQuery("github.chatroom")
+    )
+  )
+
+  /**
+   * Treats the query inputted by a user as a lucene query
+   *
+   * @param queryString the query inputted by user
+   * @return the elastic query definition
+   */
+  private def luceneQuery(queryString: String): QueryDefinition = {
+    stringQuery(
+      replaceFields(queryString)
```

Comment:
Isn’t this vulnerable to code injection attacks?

Reply:
Par exemple, si je clique sur `typelevel` depuis la page de cats
![lucene-query](https://user-images.githubusercontent.com/13123162/77640448-b4d7c200-6f5a-11ea-8cf7-be475b9349ad.jpeg)

Reply:
OK, thanks for the clarification.